### PR TITLE
A4A > Partner Directory: Section activation with the a4a-partner-directory feature flag

### PR DESCRIPTION
**Pull request by cleacos**
Resolves: https://github.com/Automattic/automattic-for-agencies-dev/issues/772
Resolves: https://github.com/Automattic/automattic-for-agencies-dev/issues/744

## Proposed Changes

This PR adds the new feature flag `a4a-partner-directory` and removes the Partner Directory section from staging.

The Partner Directory section is available in these contexts:

- **Development:** always
- **Horizon:** always
- **Staging:** with feature flag
- **Production:** The section is deactivated for any case.

## Testing Instructions

- Check the code
- Ensure the feature flag on each context is set correctly

- Go to the GH issue and follow the rest of the testing instructions there.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: config/a8c-for-agencies-stage.json [hunk 1]
```diff
@@ -52,7 +53,7 @@
 		"a8c-for-agencies-signup": true,
 		"a8c-for-agencies-referrals": true,
 		"a8c-for-agencies-settings": false,
-		"a8c-for-agencies-partner-directory": true,
+		"a8c-for-agencies-partner-directory": false,
```
Comment (yashwin) [inline config/a8c-for-agencies-stage.json:56]: Should we get rid of this section flag & use the feature flags at these 2 places:

https://github.com/Automattic/wp-calypso/blob/58e0a8bad04668f41ccdaa3cc77f02d85f91f697/client/a8c-for-agencies/hooks/use-onboarding-tours.ts#L88

https://github.com/Automattic/wp-calypso/blob/58e0a8bad04668f41ccdaa3cc77f02d85f91f697/client/a8c-for-agencies/sections/overview/partner-directory-onboarding-card/hooks/use-partner-directory-onboard-card.ts#L21

**Review by yashwin — APPROVED** [review]
It works as expected & the code changes look good. 

I left a couple of comments. Let me know what you think.

File: config/a8c-for-agencies-stage.json (hunk 1 above)
Comment (cleacos) [reply config/a8c-for-agencies-stage.json:56]: Good catch! Yes

**Review by yashwin — APPROVED** [review]
LGTM!

**Pull request merged**